### ISSUE TITLE
@cypress/vue Expose some of the internal types to allow for easier extending

### ISSUE THREAD

**Issue by LolliDepp** [cypress-io/cypress#14752]
### What would you like?
Most of the types internally used by `@cypress/vue` are not exported - I would like these to be exported.

### Why is this needed?

In my project I'm using Vue + TSX, and the library `vue-tsx-support` allows me to enforce types for props, slots and similar.

In my index.d.ts file, I am extending the definition of `Mount` and `MountCallback` as follows:
```js
type OptionsOrProps<T> = Partial<
    {
      propsData?: tsx.TsxComponentAttrs<T>;
    } & MountOptions &
      VueTestUtilsConfigOptions
  >;

  export type Mount = <TProps, TEvents = {}, TSlots = {}>(
    component: { new (): tsx.Component<TProps, TEvents, TSlots> },
    optionsOrProps?: OptionsOrProps<TProps>
  ) => Cypress.Chainable<Window>;
  export type MountCallback = <TProps, TEvents = {}, TSlots = {}>(
    component: { new (): tsx.Component<TProps, TEvents, TSlots> },
    optionsOrProps?: OptionsOrProps<TProps>
  ) => () => Cypress.Chainable<Window>;
```

This use of the generic `TProps` + tsx support package allowed me to get type checking on the value I'm passing to `propsData` to ensure it's the type expected by the component.

I recently updated from `cypress-vue-unit-test` v1.4.0 (where there weren't many typings) to the latest of `@cypress/vue` (were you have it all nice and typed).
While previously I had manually listed the few properties that were supported, now the list is quite extensive, so what I found myself doing was copying your declarations for `MountOptions ` into my own index.d.ts, to make them available.

So, bottom line:
if `MountOptions` type was exported by `@cypress/vue`, then I could just import it when applying my extension to Mount/MountCallback, while right now I have instead to copy the full declarations manually - and this is prone to break as your package gets updated...

Thank you!
